File: Anyelo Carrillo Java/Listas.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class listas {
    public static void main(String[] args) {
        // Crear una lista de días de la semana
        List<String> lista = new ArrayList<>(Arrays.asList("Lunes", "Martes", "Miercoles", "Jueves", "viernes", "sabado"));

        // Imprimir el elemento en la posición 4 (índice 4)
        System.out.println(lista.get(4)); // Imprime "viernes"

        // Imprimir la lista completa
        System.out.println(lista); // Imprime toda la lista

        // Imprimir una sublista desde el índice 0 hasta el 3 (exclusivo)
        System.out.println(lista.subList(0, 3)); // Imprime ["Lunes", "Martes", "Miercoles"]

        // Crear una lista heterogénea (mezcla de tipos)
        List<Object> listaHeterogenea = new ArrayList<>(Arrays.asList(
            "Lunes", "Martes", "Miercoles", "Jueves", "viernes", "sabado", 1, 2, 3,
            Arrays.asList("Esteban", 0.2, 2.25, true)
        ));

        // Imprimir una sublista desde el índice 0 hasta el 4 (exclusivo)
        System.out.println(listaHeterogenea.subList(0, 4)); // Imprime ["Lunes", "Martes", "Miercoles", "Jueves"]

        // Imprimir una sublista de la lista anidada en la posición 9
        List<Object> subLista = (List<Object>) listaHeterogenea.get(9); // Obtener la lista anidada
        System.out.println(subLista.subList(0, 3)); // Imprime ["Esteban", 0.2, 2.25]
    }
}

File: Anyelo Carrillo Go/Listas.go
package main

import "fmt"

func main() {
	// Crear una lista (slice) de strings
	Lista := []string{"Lunes", "Martes", "Miercoles", "Jueves", "viernes", "sabado"}

	// Acceder al elemento en la posición 4 (índice 4)
	fmt.Println(Lista[4]) // Imprime "viernes"

	// Imprimir la lista completa
	fmt.Println(Lista)

	// Obtener un subconjunto de la lista (desde el índice 0 hasta el 3, excluyendo el 3)
	fmt.Println(Lista[0:3]) // Imprime ["Lunes", "Martes", "Miercoles"]

	// Crear una lista heterogénea (en Go, esto no es directamente posible, se usa una estructura o interface{})
	// Para simular una lista heterogénea, se puede usar []interface{}
	ListaHeterogenea := []interface{}{
		"Lunes", "Martes", "Miercoles", "Jueves", "viernes", "sabado", 1, 2, 3,
		[]interface{}{"Esteban", 0.2, 2.25, true},
	}

	// Acceder a los primeros 4 elementos y a un subconjunto de la lista anidada
	fmt.Println(ListaHeterogenea[0:4], ListaHeterogenea[9].([]interface{})[0:3])
}
